test/jdk/com/sun/net/httpserver/BasicAuthToken.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * @test
+ * @bug 8144100
+ * @run main/othervm BasicAuthToken
+ * @summary checking token sent by client should be done in case-insensitive manner
+ */
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.util.Base64;
+import com.sun.net.httpserver.Authenticator;
+import com.sun.net.httpserver.BasicAuthenticator;
+import com.sun.net.httpserver.HttpServer;
+
+public class BasicAuthToken {
+    static final String CRLF = "\r\n";
+    static final String someContext = "/test";
+
+    public static void main(String[] args) throws Exception {
+        HttpServer server = server();
+        try {
+            client(server.getAddress().getPort());
+        } finally {
+            server.stop(0);
+        }
+    }
+
+    static HttpServer server() throws Exception {
+        String realm = "someRealm";
+        ServerAuthenticator authenticator = new ServerAuthenticator(realm);
+        HttpServer server = HttpServer.create(new InetSocketAddress(0), 0);
+        server.createContext(someContext, exchange -> {
+            if (authenticator.authenticate(exchange) instanceof Authenticator.Failure) {
+                exchange.sendResponseHeaders(401, 0);
+                exchange.close();
+                return;
+            }
+            exchange.sendResponseHeaders(200, 1);
+            exchange.close();
+        }).setAuthenticator(authenticator);
+        server.start();
+        return server;
+    }
+
+    static void client(int port) throws Exception {
+        try (Socket socket = new Socket("localhost", port)) {
+            BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(socket.getOutputStream()));
+            BufferedReader reader = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+            String credentials = "username:password";
+            String encodedCredentials = Base64.getEncoder().encodeToString(credentials.getBytes());
+            writer.write("GET " + someContext + " HTTP/1.1" + CRLF);
+            writer.write("Host: localhost:" + port + CRLF);
+            writer.write("User-Agent: Java/" + System.getProperty("java.version") + CRLF);
+            writer.write("Authorization: BAsIc " + encodedCredentials + CRLF);
+            writer.write(CRLF);
+            writer.flush();
+
+            System.err.println("Server response");
+            String statusLine = reader.readLine();
+            System.err.println(statusLine);
+
+            if (statusLine.startsWith("HTTP/1.1 401")) {
+                throw new RuntimeException("Basic Authentication failed: case-insensitive token" +
+                        " used to identify authentication scheme  sent by client parsed incorrectly");
+            }
+        }
+    }
+
+    static class ServerAuthenticator extends BasicAuthenticator {
+        ServerAuthenticator(String realm) {
```

Comment:
Since this test relies on the authenticator being invoked, just as an additional check, you might want to have a boolean field in this class (named something like `invoked`) which will be set to `true` only when the `checkCredentials` gets called. Then after checking the response code in the test, you can additionally assert that this field's value is `true`, just to be certain that this authenticator was indeed invoked.

Reply:
Fixed, thanks